### PR TITLE
Update wc-page-functions.php

### PR DESCRIPTION
**Pull request by pwsonline**
### Changes proposed in this Pull Request:
PHP 8.1 needs cast on needle and haystack if they can be null.

### CODE REVIEW

**Comment by Konamiman** [comment]
Hi @pwsonline, thanks for your contribution. Could you please add testing instructions to the pull request description? What action would fail in PHP 8.1 without your fix but will work with the fix? Thank you!

**Comment by pwsonline** [comment]
> Hi @pwsonline, thanks for your contribution. Could you please add testing instructions to the pull request description? What action would fail in PHP 8.1 without your fix but will work with the fix? Thank you!

Hello there,
No action fails, it just throws a PHP Warning (log and/or displayed), my fix - with the knowhow of php.net - eliminates that warning.

**Comment by Konamiman** [comment]
@pwsonline Ok, but what's what needs to be done exactly to see the warnings? Maybe load some specific page?

Also could you please add a changelog file? You can do so by running `pnpm --filter=plugins/woocommerce run changelog add`.

**Comment by Joh84** [comment]
> @pwsonline Ok, but what's what needs to be done exactly to see the warnings? Maybe load some specific page?

If customer logout is filled under woocommerce-settings-advanced-endpoints, user is logged out and a menu is specified this happens on every page if not logged in. Workaround for now is removing the filter `remove_filter( 'wp_nav_menu_objects', 'wc_nav_menu_items', 10 );`

**Comment by pwsonline** [comment]
> @pwsonline Ok, but what's what needs to be done exactly to see the warnings? Maybe load some specific page?
> 

Joh84 answered that part.

> Also could you please add a changelog file? You can do so by running `pnpm --filter=plugins/woocommerce run changelog add`.

As I was creating this PR on the web interface, not sure where I could create the changelog.

**Comment by berzley** [comment]
Issue is a logged-out user receives a deprecation notice for each menu item when the nav item URL contains no query parameter(s).

This occurs because in PHP 8.1 passing null to [`strstr`](https://www.php.net/manual/en/function.strstr.php) was deprecated. Looking at documentation, this issue may have been introduced in PHP 8.0.

[`wp_parse_url`](https://developer.wordpress.org/reference/functions/wp_parse_url/) returns `null` if the URL component does not exist.

How about we just check `$path` and `$query` are valid before calling `strstr`?
```php
if ( ( $path && strstr( $path, $customer_logout ) ) || ( $query && strstr( $query, $customer_logout ) ) ) {
	unset( $items[ $key ] );
}
```

PHP 8.1 is 6-months away from losing active support, and PHP 8.0 will be entering end of life. This is overdue.

File: plugins/woocommerce/includes/wc-page-functions.php [hunk 1]
```diff
@@ -133,8 +133,8 @@ function wc_nav_menu_items( $items ) {
 				if ( empty( $item->url ) ) {
 					continue;
 				}
-				$path  = wp_parse_url( $item->url, PHP_URL_PATH );
-				$query = wp_parse_url( $item->url, PHP_URL_QUERY );
+				$path  = (string)wp_parse_url( $item->url, PHP_URL_PATH );
```
Comment (rrennick) [inline plugins/woocommerce/includes/wc-page-functions.php]: Using a null coelesce will work for both here

```suggestion
				$path  = wp_parse_url( $item->url, PHP_URL_PATH ) ?? '';
```

**Review by Konamiman — APPROVED** [review]

**Comment by mattish91** [comment]
Have there been any update to this issue at all? 

We are currently running php8.1 (Klarna Payment Gateway requirement) but WooCommerce seasm to be filling our server storage with error logs regarding wc-page-functions.php line 139 as someone above refered to. 

It's been like this for some time now and i have contacted Klarna about it that says the update was a requirement of WooCommerce. Obviously WooCommerce is still working with php8.0 but not php8.1 and a downgrade for Klarna payment Gateway doesn't seam to be an option as it wouldn't work with older versions. 

I have been looking at peoples comments with the same issue that have rewritten line 139, but that clearly doesn't work either as the site won't load at all after those changes. 

In the current state, everything seams to show correctly (for the customers) except for the huge error logs that gets generated by every page visit, refresh and bot coming in contact with our production site. 

Is there any solution to this? Please point me in the correct direction as this needs a solution soon.

**Comment by Konamiman** [comment]
Hi @mattish91, this pull request has been merged into `trunk` and will be part of WooCommerce 7.9, to be released in a few days.

**Pull request merged**